cmds/exp/netbootxyz/network.go

```diff
@@ -14,15 +14,82 @@ import (
 	"net"
 	"net/http"
 	"os"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/u-root/u-root/pkg/dhclient"
 	"github.com/vishvananda/netlink"
 )
 
-func downloadFile(filepath string, url string) error {
+// Counts number of bytes written
+// Conforms to io.Writer interface
+type ProgressCounter struct {
+	Downloaded    uint64
+	Total         uint64
+	PreviousRatio int
+}
```

Comment:
could you add:
W io.Writer 
so we are not tied to os.Stdout?

Reply:
Hmm, I don't know what you mean, sorry, I've barely written any Go.

Reply:
I guess what was meant is sth like following:

type ProgressCounter struct {
	Downloaded    uint64
	Total         uint64
	PreviousRatio int
        W io.Writer
}

so you implementation writes to W instead of assuming / writting to Stdout.

You can still achieves same default behavior to Stdout, by giving os.Stdout to W in line 108

`counter := &ProgressCounter{Total: size, PreviousRatio: 0, W: os.Stdout}`

The benefit is a flexibility to pass in a different writer to W

Reply:
Oh I see, I thought it would involve making the struct conform the the Writer interface.

Done. Tested, still works.